apps/quarkus-full-microprofile/src/main/java/com/example/quarkus/secure/JWTResource.java

```diff
@@ -0,0 +1,85 @@
+package com.example.quarkus.secure;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.JWTOptions;
+import io.vertx.ext.auth.PubSecKeyOptions;
+import io.vertx.ext.auth.jwt.JWTAuth;
+import io.vertx.ext.auth.jwt.JWTAuthOptions;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Response;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.UUID;
+
+@Path("/secured")
+@ApplicationScoped
+public class JWTResource {
+
+    private String key;
+
+    @PostConstruct
+    public void init() {
+        try (final InputStream is = Objects.requireNonNull(JWTResource.class.getResourceAsStream("/privateKey.pem"))) {
+            key = new String(is.readAllBytes(), StandardCharsets.US_ASCII);
+        } catch (Exception e) {
+            key = null;
+        }
+    }
+
+    @ConfigProperty(name = "quarkus.http.port")
+    int port;
+
+    @GET
+    @Path("/test")
+    public String testSecureCall() {
+        if (key == null) {
+            throw new WebApplicationException("Unable to read privateKey.pem", 500);
+        }
+        // Why not try-with-resources, right?
+        // Because with certain older versions, Client is not auto-closable.
```

Comment:
We should document which older versions still need it so that we can clean this up once we are past that.

Reply:
Will add comment.